File: data/imported/knowledge/3d/typescript/merged/snippets/snippet_029.ts
it('testStarShapeHighlyConcave', () => {
        const starMesh = new Mesh('star');
        starMesh.controlPoints = [
            new Vector4(0, 1, 0, 1),
            new Vector4(0.3, 0.3, 0, 1),
            new Vector4(1, 0.3, 0, 1),
            new Vector4(0.4, -0.1, 0, 1),
            new Vector4(0.5, -0.6, 0, 1),
            new Vector4(0.2, -0.2, 0, 1),
            new Vector4(-0.5, -0.6, 0, 1),
            new Vector4(-0.1, -0.1, 0, 1),
            new Vector4(-1, 0.3, 0, 1),
            new Vector4(-0.3, 0.3, 0, 1),
        ];
        starMesh.createPolygon([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]);

        expect(starMesh.polygonCount).toBe(1);

        const triangulated = PolygonModifier.triangulate(starMesh);
        
        expect(triangulated.polygonCount).toBe(8);
    })

File: data/imported/knowledge/3d/python/merged/snippets/snippet_029.py
# Adapted from aspose.org: knowledge/3d/python/merged/snippets/snippet_029.py @ 7f72da4e1423546104b40fa8cebf5b9ae3ce9c91
# Imported under the authorization recorded in
# plans/investigations/evidence/imported-corpus-v1/licensing-resolution-state.md

def test_export_lambert_material(self):

        scene = Scene()

        mesh = Mesh('TestMesh')



        mesh._control_points.append(Vector4(0.0, 0.0, 0.0, 1.0))

        mesh._control_points.append(Vector4(1.0, 0.0, 0.0, 1.0))

        mesh._control_points.append(Vector4(0.0, 1.0, 0.0, 1.0))

        mesh.create_polygon(0, 1, 2)



        material = LambertMaterial('BlueMaterial')

        material.diffuse_color = Vector3(0.0, 0.0, 1.0)



        node = scene.root_node.create_child_node('TestNode')

        node.entity = mesh

        node.material = material



        stream = io.BytesIO()

        options = ColladaSaveOptions()

        options.file_name = 'test.dae'



        from aspose.threed.formats.collada.ColladaExporter import ColladaExporter

        exporter = ColladaExporter()

        exporter.export(scene, stream, options)



        stream.seek(0)

        content = stream.read().decode('utf-8')



        root = ET.fromstring(content)



        library_effects = root.find('{http://www.collada.org/2005/11/COLLADASchema}library_effects')

        self.assertIsNotNone(library_effects)



        effect = library_effects.find('{http://www.collada.org/2005/11/COLLADASchema}effect')

        self.assertIsNotNone(effect)



        profile_common = effect.find('{http://www.collada.org/2005/11/COLLADASchema}profile_COMMON')

        self.assertIsNotNone(profile_common)



        technique = profile_common.find('{http://www.collada.org/2005/11/COLLADASchema}technique')

        self.assertIsNotNone(technique)



        lambert = technique.find('{http://www.collada.org/2005/11/COLLADASchema}lambert')

        self.assertIsNotNone(lambert)